mysql-test/main/opt_trace_load_stats_innodb.result

```diff
@@ -0,0 +1,444 @@
+          "access_type": "ALL",
+          "possible_keys": ["c1"],
+          "loops": 1,
+          "rows": 10000,
+          "cost": 1.6605152,
+          "filtered": 100,
+          "attached_condition": "tt1.c1 is not null"
+        }
+      },
+      {
+        "table": {
+          "table_name": "tt2",
+          "access_type": "ref",
+          "possible_keys": ["c1"],
+          "key": "c1",
+          "key_length": "5",
+          "used_key_parts": ["c1"],
+          "ref": ["db1.tt1.c1"],
+          "loops": 10000,
+          "rows": 100,
+          "cost": 1001.735053,
+          "filtered": 100
+        }
+      }
+    ]
+  }
+}
+set @saved_explain_output=@explain_output;
+set optimizer_replay_context="";
+truncate table t1;
+set optimizer_replay_context=@saved_opt_context;
+set @explain_output='$explain_output';
+select JSON_EQUALS(@saved_explain_output, @explain_output);
+JSON_EQUALS(@saved_explain_output, @explain_output)
+1
+set optimizer_replay_context="";
+#
+# On the same table, test with non-constant ref access with out using an index
+# Also, re-insert data
+#
+insert into t1 select seq%100, seq%500 from seq_1_to_10000;
+analyze table t1;
+Table	Op	Msg_type	Msg_text
+db1.t1	analyze	status	Engine-independent statistics collected
+db1.t1	analyze	status	OK
+set optimizer_replay_context="";
+explain format=json select * from t1 as tt1, t1 as tt2 where tt1.c2 = tt2.c2 and tt1.c2 <= 100
+set @trace= (select trace from information_schema.optimizer_trace);
+set @saved_opt_context=
+(select json_pretty(json_extract(
+json_extract(@trace, "$**.optimizer_context"),
+'$[0]'
+                      )
+));
+set @explain_output='$explain_output';
+select @explain_output;
+@explain_output
+{
+  "query_block": {
+    "select_id": 1,
+    "cost": 374.9067624,
+    "nested_loop": [
+      {
+        "table": {
+          "table_name": "tt1",
+          "access_type": "ALL",
+          "loops": 1,
+          "rows": 10000,
+          "cost": 1.6605152,
+          "filtered": 20,
+          "attached_condition": "tt1.c2 <= 100"
+        }
+      },
+      {
+        "block-nl-join": {
+          "table": {
+            "table_name": "tt2",
+            "access_type": "ALL",
+            "loops": 2000,
+            "rows": 10000,
+            "cost": 373.2462472,
+            "filtered": 20
+          },
+          "buffer_type": "flat",
+          "buffer_size": "21Kb",
+          "join_type": "BNL",
+          "attached_condition": "tt2.c2 = tt1.c2"
+        }
+      }
+    ]
+  }
+}
+set @saved_explain_output=@explain_output;
+set optimizer_replay_context="";
+truncate table t1;
+set optimizer_replay_context=@saved_opt_context;
+set @explain_output='$explain_output';
+select JSON_EQUALS(@saved_explain_output, @explain_output);
+JSON_EQUALS(@saved_explain_output, @explain_output)
+1
+set optimizer_replay_context="";
+drop table t1;
+#
+# Query on a single table having 1 index.
+# However, a constant eq_ref access is used on the index.
```

Comment:
It's not eq_ref, it's ref.
(And I think it doesn't make sense to try adding a testcase  with eq_ref(cost)).